R/finalize_plot.R

```diff
@@ -434,3 +463,75 @@ finalize_plot <- function(input_plot = NULL,
     return(final_plot)
   }
 }
+
+
+
+#' @noRd
+# Function to create plot object with left aligned legend on top
+buildChart <- function(input_plot,
+                       plot_constants,
+                       legend_build) {
+
+  # in safe mode, don't extract legend
+  if(legend_build == "safe"){
+    return(ggplotGrob(input_plot))
+  }
+
+  # in no legend mode, remove legend altogether
+  if(legend_build == "none"){
+    output_plot <- input_plot + theme(legend.position = "none")
+    return(ggplotGrob(output_plot))
+  }
+
+  # Reformat plot
+  ## IS ANY OF THIS ACTUALLY NECESSARY? CAN WE MOVE MORE OF THIS OVER TO THEME_CMAP?
+  format_plot <- input_plot + theme(
+    # ensure legend is left aligned
+    legend.position = "left",
```

Comment:
This is needed to force multiple legends to stack vertically but is a fairly implicit/indirect way to do so. We should look into whether it is possible to do this more directly. If possible, this should be done at the `theme_cmap()` level so that it applies not only in `legend_build = "adjust"` mode. If it is possible, we should look into easy opportunities for customization.

Reply:
added legend.box.just = "left" to theme_cmap in commit 7cec129. This seems to have fixed the problem